### ISSUE TITLE
@requires alternative to mark a dependency without removing it from the main docs tree

### ISSUE THREAD

**Issue by snakemastr** [vue-styleguidist/vue-styleguidist#953]
So @requires marks another component as a dependency but removes it from the main docs.
Is it possible to do so without removing it?

In the current usecase I have a Modal component which is used inside another component. 
I would love to refer in the documents to the Modal component without removing it as a component on it's own.

Something like 
`@uses './Modal'`

Would be cool if it would work with interfaces also.

**Comment by elevatebart** [comment]
Hello @snakemastr,

That is an interesting idea.
What would this tag look like in the documentation? 
A link to the other components documentation? 

Where would you expect this link to be? Before or after the component props and doc?

**Comment by snakemastr** [comment]
> A link to the other components documentation?

Yes I think the link to the component itself would be logical as it could refer to a component in a different section or on a different page.

> What would this tag look like in the documentation? 
> Where would you expect this link to be? Before or after the component props and doc?

I would place it in the same position in the hierarchy as "requires" so dependencies are grouped. I made an example for the combination with requires.
Perhaps with lesser space inbetween as they could be presented together
![Knipsel_1](https://user-images.githubusercontent.com/18302945/91698230-ac6f5d00-eb72-11ea-990f-ad4fb7943691.PNG)

A more compact view 
![Knipsel_2](https://user-images.githubusercontent.com/18302945/91698233-ada08a00-eb72-11ea-9fd6-dbde5d1c4495.PNG)

**Comment by elevatebart** [comment]
I remember @davatron5000 talking about this kind of feature being useful in Austin TX. 
We might even be able to go one step further and automatically extract which components are used from the code.

The `@use` tag is a rendering of a link. 
One non-obvious part of the making of this feature is figuring out the content of the href. 
Each component can have url inside of a section so resolving the path in the sections means probably having all 
component filePath => component docs urlPath map available at render time.
Since we render the side bar for all the sections, I guess we could find a way to store this structure.
Then it would be a matter of waiting for this structure to be ready to pick in the map for the url.

I will create another issue for the docgen extraction.
